include/deal.II/physics/elasticity/kinematics.h

```diff
@@ -0,0 +1,365 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__elasticity__kinematics_h
+#define dealii__elasticity__kinematics_h
+
+
+#include <deal.II/base/symmetric_tensor.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/physics/elasticity/standard_tensors.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Physics
+{
+
+  namespace Elasticity
+  {
+
+    /**
+     * A collection of tensor definitions for deformation and strain measures,
+     * as well as a few special transformations, that conform to notation used in
+     * standard scientific literature. These hold specifically for the codimension
+     * 0 case, where the metric tensor is the identity tensor.
+     *
+     * @author Jean-Paul Pelteret, Andrew McBride, 2016
+    */
+    namespace Kinematics
+    {
+
+      /**
+       * @name Deformation tensors
+       */
+//@{
+
+      /**
+       * Returns the deformation gradient Tensor,
+       * as constructed from the material displacement gradient
+       * Tensor @p Grad_u.
+       * The result is expressed as
+       * @f[
+       *  \mathbf{F} := \mathbf{I} + \nabla_{0}\mathbf{u}
+       * @f]
+       * where $\mathbf{u} = \mathbf{u}(\mathbf{X})$ is the displacement at position
+       * $\mathbf{X}$ in the referential configuration.
+       * The differential operator $\nabla_{0}$ is defined as
+       * $\frac{\partial}{\partial \mathbf{X}}$.
+       *
+       * @param[in] Grad_u Material displacement gradient tensor $\nabla_{0}\mathbf{u}$
+       * @return    The deformation gradient tensor $\mathbf{F}$
+       */
+      template <int dim, typename Number>
+      Tensor<2, dim, Number>
+      F (const Tensor<2, dim, Number> &Grad_u);
+
+      /**
+       * Returns the isochoric counterpart of the deformation gradient Tensor,
+       * deformation gradient Tensor @p F itself.
```

Comment:
here and in the next one, i am puzzled with 

> ...the deformation gradient Tensor, deformation gradient Tensor @p F itself.

maybe it's just a long day...